pkg/controller/installation/products/mobiledeveloperconsole/reconciler.go

```diff
@@ -0,0 +1,246 @@
+			for j, env := range container.Env {
+				if env.Name == "OPENSHIFT_HOST" {
+					dc.Spec.Template.Spec.Containers[i].Env[j].Value = inst.Spec.MasterURL
+					break
+				}
+			}
+		}
+	}
+	if err := client.Update(ctx, dc); err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to update dc for mdc")
+	}
+
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      routeResourceName,
+			Namespace: r.Config.GetNamespace(),
+		},
+	}
+
+	if err := client.Get(ctx, pkgclient.ObjectKey{Name: route.Name, Namespace: route.Namespace}, route); err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "could not read mdc route")
+	}
+
+	var url string
+	if route.Spec.TLS != nil {
+		url = fmt.Sprintf("https://" + route.Spec.Host)
+	} else {
+		url = fmt.Sprintf("http://" + route.Spec.Host)
+	}
+	if r.Config.GetHost() != url {
+		r.Config.SetHost(url)
+		r.ConfigManager.WriteConfig(r.Config)
+	}
+
+	phase, err := r.ReconcileOauthClient(ctx, inst, &oauthv1.OAuthClient{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: inst.Spec.NamespacePrefix + string(r.Config.GetProductName()),
+		},
+		Secret: clientSecret,
+		RedirectURIs: []string{
+			r.Config.GetHost(),
+		},
+		GrantMethod: oauthv1.GrantHandlerAuto,
+	}, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	// if there are no errors, the phase is complete
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) getOauthClientSecret(ctx context.Context, serverClient pkgclient.Client) (string, error) {
+
+	oauthClientSecrets := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: r.ConfigManager.GetOauthClientsSecretName(),
+		},
+	}
+
+	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: oauthClientSecrets.Name, Namespace: r.ConfigManager.GetOperatorNamespace()}, oauthClientSecrets)
+	if err != nil {
+		return "", errors.Wrapf(err, "Could not find %s Secret", oauthClientSecrets.Name)
+	}
+
+	clientSecretBytes, ok := oauthClientSecrets.Data[string(r.Config.GetProductName())]
+	if !ok {
+		return "", errors.Wrapf(err, "Could not find %s key in %s Secret", string(r.Config.GetProductName()), oauthClientSecrets.Name)
+	}
+	return string(clientSecretBytes), nil
+}
+
+func (r *Reconciler) handleProgress(ctx context.Context, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	r.logger.Info("checking mdc pods are running")
+
+	pods := &corev1.PodList{}
+	err := client.List(ctx, &pkgclient.ListOptions{Namespace: r.Config.GetNamespace()}, pods)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to list mdc in mdc namespace")
+	}
+
+	//and they should all be ready
+checkPodStatus:
+	for _, pod := range pods.Items {
+		for _, cnd := range pod.Status.Conditions {
+			if cnd.Type == v1.ContainersReady {
+				if cnd.Status != v1.ConditionStatus("True") {
+					return v1alpha1.PhaseInProgress, nil
+				}
+				break checkPodStatus
+			}
+		}
+	}
+
+	r.logger.Infof("all pods ready, mdc complete")
```

Comment:
Can we check MDC StatusPhase [here](https://github.com/aerogear/mobile-developer-console-operator/blob/master/pkg/apis/mdc/v1alpha1/mobiledeveloperconsole_types.go#L57) to check if all MDC resources have completed provisioning?